### ISSUE TITLE
have_enqueued_email not working

### ISSUE THREAD

**Issue by coding-bunny** [rspec/rspec-rails#2287]
### Subject of the issue
I'm trying to use the new `have_enqueued_email` matcher in one of my specs, but RSpec keeps telling me the format is incorrect

### Your environment
* Ruby version: 2.6.5
* rspec-expectations version: 3.9.0

### Steps to reproduce
- Write code to schedule a Mailer with `.deliver_now`
- Setup RSpec to support ActiveJob
- write the following spec : `expect { post(:email_quote, params: quote_params) }.to have_enqueued_email(MyMailer, :quote)`

### Expected behavior
Test to pass

### Actual behavior
Test fails with the error : 
```
You must pass an argument rather than a block to `expect` to use the provided matcher (have enqueued email QuoteMailer, :quote), or the matcher must implement `supports_block_expectations?`.
```

The test:
```
describe "#quote_email" do
    subject(:quote_email) { post(:email_quote, params: quote_params) }
    let(:email) { "test@example.com" }
    let(:quote_params) { { item: item_params, email: email } }
    let(:item_params) do
      { variant_id: variant.id, quantity: 100, decoration_id: decoration.id, number_of_colors: 1, is_sample: false }
    end

    before { ActiveJob::Base.queue_adapter = :test }

    it "properly schedules the Mailer for sending the email" do
      expect { subject }.to have_enqueued_email(QuoteMailer, :quote)
    end
  end
```

The code for invoking in the controller:
```
def email_quote
    item = params.require(:item).permit!.to_h
    email = params.require(:email)

    QuoteMailer.quote(item, email).deliver_later
    render json: true
  end
```

**Comment by coding-bunny** [comment]
I'm not familiar with Pry, so I tried the command you told to use:

```
> have_enqueued_mail
=> #<RSpec::Matchers::BuiltIn::Has:0x00007f8e87a4e558 @args=[], @block=nil, @method_name=:have_enqueued_mail>
```

**Comment by coding-bunny** [comment]
```
0> have_enqueued_email(QuoteMailer, :quote).supports_block_expectations?
=> false
```

**Comment by JonRowe** [comment]
> ```
> > have_enqueued_mail
> => #<RSpec::Matchers::BuiltIn::Has:0x00007f8e87a4e558 @args=[], @block=nil, @method_name=:have_enqueued_mail>
> ```
This is your problem you're not using the right matcher, thats the `has_<n>` matcher which is not the same. Are you using rspec-rails from Github or from the `4.0.0.beta3` or `4.0.0.beta4`?

**Comment by coding-bunny** [comment]
normal gem installation

```
group :test do
  gem "factory_bot_rails", require: false
  gem "rails-controller-testing"
  gem "rspec-collection_matchers"
  gem "rspec-nc"
  gem "rspec-rails"
  gem "rspec_junit_formatter"
  gem "simplecov", require: false
  gem "timecop"
end
```

**Comment by JonRowe** [comment]
Ok, you need to use `4.0.0.beta3` or `4.0.0.beta4` to use this matcher. Rubygems will not give you beta's unless you ask for them, change  `gem "rspec-rails"` for `gem "rspec-rails", "4.0.0.beta4"`

**Comment by pirj** [comment]
Also, I suggest moving `rspec-rails` to `:development, :test` group [as doc suggests](https://github.com/rspec/rspec-rails#installation).

**Comment by pirj** [comment]
Some background https://github.com/rspec/rspec-rails/issues/2200#issuecomment-549389317 https://github.com/rspec/rspec-rails/pull/2117

**Comment by JonRowe** [comment]
I'm not sure thats relevant @pirj, the matcher didn't exist before 4.0.0 you'd need a beta anyway.

**Comment by coding-bunny** [comment]
Thanks for the clarifications. I think I just got lost in all the docs I was reading about this matcher, that I missed the point this isn't released yet officially.

**Comment by coding-bunny** [comment]
this is what threw me off : https://www.rubydoc.info/gems/rspec-rails/RSpec%2FRails%2FMatchers:have_enqueued_mail

As the version on top says 3.9.0

**Comment by JonRowe** [comment]
It is officially released, we just haven't published as `4.0.0` final yet.

**Comment by JonRowe** [comment]
My apologies, it seems it was shipped in `3.9.0` I mis-remembered and then mis-read the change log :joy: , yes there is a bug with the require, you can work around it by requiring it manually.

**Comment by JonRowe** [comment]
I've release 3.9.1 as a fix for this, it seems silly not to.
